Configure

```diff
@@ -1415,9 +1415,11 @@ if (!$disabled{makedepend}) {
         # For VC- and vms- targets, there's nothing more to do here.  The
         # functionality is hard coded in the corresponding build files for
         # cl (Windows) and CC/DECC (VMS).
-    } elsif ($predefined{__GNUC__} >= 3) {
+    } elsif ($predefined{__GNUC__} >= 3 && !(defined($predefined{__APPLE_CC__})
+                                             && !defined($predefined{__clang__}))) {
```

Comment:
Here I'm going to insist on omitting defined(). Because it's only done when you can't remove anything, and you can remove defined(). (Yeah, I care more about 1.1.x :-) And while we are at it, since we use `use warnings` in 1.1.x, it's only appropriate to address potential warning in uncommon case of non-GNU or GNU-alike with `($predefined{__GNUC__} // -1) >= 3`. Comment naturally applies even to #6167.

Reply:
So, the conditional should be:
```
elsif (($predefined{__GNUC__} // -1) >= 3 && !($predefined{__APPLE_CC__} && !$predefined{__clang__}))
```
or
```
elsif (($predefined{__GNUC__} // -1) >= 3 && !(($predefined{__APPLE_CC__} // 0) && !($predefined{__clang__} // 0)))
```
?? 
(Never used the // operator before...)

Reply:
First one. Expression that evaluates to undefined can be used in logical condition, while arithmetic condition would trigger a warning. So `if ($predefined{__APPLE_CC__})` is fine, while `if ($predefined{__APPLE_CC__} > 5000)` triggers warning. defined() would have to be used if 0 was a possible value, but it's not, not for pre-defines in question.

Reply:
Gotcha